plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -865,23 +867,30 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
                         tableMetadatas.add(TableColumnsMetadata.forTable(tableName, columns));
                     });
 
-                    for (SchemaTableName tableName : remainingTables) {
-                        try {
-                            Table icebergTable = catalog.loadTable(session, tableName);
-                            List<ColumnMetadata> columns = getColumnMetadatas(icebergTable.schema(), typeManager);
-                            tableMetadatas.add(TableColumnsMetadata.forTable(tableName, columns));
-                        }
-                        catch (TableNotFoundException e) {
-                            // Table disappeared during listing operation
-                        }
-                        catch (UnknownTableTypeException e) {
-                            // Skip unsupported table type in case that the table redirects are not enabled
-                        }
-                        catch (RuntimeException e) {
-                            // Table can be being removed and this may cause all sorts of exceptions. Log, because we're catching broadly.
-                            log.warn(e, "Failed to access metadata of table %s during streaming table columns for %s", tableName, prefix);
-                        }
-                    }
+                    List<CompletableFuture<Void>> futures = remainingTables.stream()
+                            .map(tableName -> CompletableFuture.runAsync(() -> {
```

Comment:
We generally don't use the common fork join pool in Trino. We should inject a dedicated executor which is bounded at some number and use that.
GET_METADATA_BATCH_SIZE is 1000, I don't think we want that many parallel fetches, something like 8 parallel fetches is probably enough.

Reply:
See the way BoundedExecutor is setup in https://github.com/trinodb/trino/pull/24110/files#diff-e72d63b77858985902ad4bb32d26e11cfaab0470c78821ce5416c2c97da4deecR222

Reply:
@raunaqmorarka Yes I was exploring how this was implemented there earlier. I’ve just updated the code. Could you please review it again? This time, I’ve incorporated the `metadataParallelism` configuration parameter and switched to using a `FixedThreadPool` for managing threads.